src/main/java/org/aerogear/digger/client/services/JobService.java

```diff
@@ -54,15 +63,24 @@ public JobWithDetails get(JenkinsServer jenkinsServer, String name) throws IOExc
      * @param name            job name that can be used later to reference job
      * @param gitRepo         git repository url (full git repository url. e.g git@github.com:digger/helloworld.git
      * @param gitBranch       git repository branch (default branch used to checkout source code)
+     * @param gitRepoCredential credential instance. See {@link Credential}.
      * @param buildParameters list of build parameters for the a parameterized job.
+     *
      * @throws IOException
      */
-    public void create(JenkinsServer jenkinsServer, String name, String gitRepo, String gitBranch, List<BuildParameter> buildParameters) throws IOException {
+    public void create(JenkinsServer jenkinsServer, String name, String gitRepo, String gitBranch, Credential gitRepoCredential, List<BuildParameter> buildParameters) throws IOException {
```

Comment:
looks like `create` does same like `update`, except their last line. Perhaps move to separate method to avoid duplicate code ?